swap_meet/vendor.py

```diff
@@ -1,2 +1,83 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        if not inventory:
+            inventory = []
+        self.inventory  = inventory
+
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+    
+
+    def get_by_id(self,id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        self.remove(my_item)
+        other_vendor.add(my_item)
+
+        other_vendor.remove(their_item)
+        self.add(their_item)
+
+        #if vendor inventory doesnt contain my item return false
+        # if vendor doesnt contain their_item return false
+        return True
+    
+    def swap_first_item(self,other_vendor):
+
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        other_vendor.add(self.inventory[0])
+
+        self.add(other_vendor.inventory[0])
+
+        self.remove(self.inventory[0])
+
+        other_vendor.remove(other_vendor.inventory[0])
+
+        return True
+    
+    def get_by_category(self,category):
+
+        self.category = category
+        
+        objects = []
+
+        for stuff in self.inventory:
+            if stuff.get_category() == category:
+                objects.append(stuff)
+
+        return objects
+    
+    # def get_best_by_category(self, category_condition):
+
+        self.category_condition = category_condition
+
+        best_condition = 0
+
+        for stuff in self.inventory(category_condition):
+            if stuff.condition_description() < best_condition:
+                best_condition = category_condition
```

Comment:
You're on the right track! Some things to think about...

1. You want to get the things from the `self.inventory` that have a specific `category_condition`. You wrote a method above this that can do that!
2. Your "best" condition is the highest one, not the lowest
3. You want to keep track of the best condition *and* the best stuff you've seen, too.

With those things in mind, see if you can come up with how to modify this code. If you still get stuck, here's a snippet that I think might help get you there: https://gist.github.com/mmcknett-ada/26444ed29c91927c05dc7d577d5bd67e